javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -260,6 +261,27 @@ void setCurrentArray() {
         }
     }
 
+    BitMappedTrie<T> filter(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+
+        final Object[] results = new Object[length()];
+        final MutableInt index = new MutableInt();
+        this.<T> visit((leaf, start, end) -> {
+            int resultIndex = index.val;
```

Comment:
I hesitated at first, but trying it out seems indeed like a good idea, thanks :)